pkg/apis/core/v1alpha1/uibutton_types.go

```diff
@@ -174,12 +187,74 @@ func (in *UIButtonList) GetListMeta() *metav1.ListMeta {
 	return &in.ListMeta
 }
 
+type UITextInputSpec struct {
+	// Initial value for this field.
+	//
+	// +optional
+	DefaultValue string `json:"defaultValue,omitempty" protobuf:"bytes,1,opt,name=defaultValue"`
+
+	// A short hint that describes the expected input of this field.
+	//
+	// +optional
+	Placeholder string `json:"placeholder,omitempty" protobuf:"bytes,2,opt,name=placeholder"`
+}
+
+type UITextInputStatus struct {
+	// The content of the text input.
+	Value string `json:"value" protobuf:"bytes,1,opt,name=value"`
+}
+
+// Defines an Input to render in the UI.
+// If UIButton is analogous to an HTML <form>,
+// UIInput is analogous to an HTML <input>.
+type UIInputSpec struct {
+	// An ID to distinguish this input from others.
```

Comment:
fwiw, I think the typical kubernetes convention would be to call this a 'Name' instead of an 'ID'. See:
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#lists-of-named-subobjects-preferred-over-maps